### ISSUE TITLE
Fails with Quarkus `2.12.0.CR1` due to `ProblemProcessor#warnOnMissingSmallryeMetricsDependency`

### ISSUE THREAD

**Issue by GavinRay97** [TietoEVRY/quarkus-resteasy-problem#223]
**Describe the bug**
```java
2022-08-17 15:46:26,104 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to start quarkus:
java.lang.RuntimeException: java.lang.IllegalArgumentException:
Build step 'com.tietoevry.quarkus.resteasy.problem.deployment.ProblemProcessor#warnOnMissingSmallryeMetricsDependency'
does not produce any build item and thus will never get executed.

Either change the return type of the method to a build item type, add a parameter of type
BuildProducer<[some build item type]>/Consumer<[some build item type]>,
or annotate the method with @Produces. Use @Produce(EmptyBuildItem.class) if you want to always execute this step.

        at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:330)                                                                                                                                     
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:252)
        at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:60)                                                                                                                 
        at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:87)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:448)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:60)
        at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:148)
        at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:103)
        at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:131)
        at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)

Caused by: java.lang.IllegalArgumentException: Build step 'com.tietoevry.quarkus.resteasy.problem.deployment.ProblemProcessor#warnOnMissingSmallryeMetricsDependency' does not produce any build item and thus will never get executed.

Either change the return type of the method to a build item type, add a parameter of type BuildProducer<[some build item type]>/Consumer<[some build item type]>, or annotate the method with @Produces. Use @Produce(EmptyBuildItem.class) if you want to always execute this step.
```

**To Reproduce**
- Sample application or code (github repo / gist / inline)
- Quarkus version, quarkus-resteasy-problem version

**Expected behavior**

**Comment by lwitkowski** [comment]
Thanks for report @GavinRay97, I'll look at it later today.

**Comment by lwitkowski** [comment]
The fix is ready in a draft PR (#224), but I'll wait for 2.12.0.Final with merging, as this change will require releasing/publishing new problem extension version.

**Comment by lwitkowski** [comment]
@GavinRay97 fyi: v2.0.1 with a fix has just been released and synced with maven central: https://repo.maven.apache.org/maven2/com/tietoevry/quarkus/quarkus-resteasy-problem/2.0.1/
